pyccl/halos/halo_model.py

```diff
@@ -579,13 +608,28 @@ def pkf(sf):
             pk_2h = 0.
 
         if get_1h:
-            pk_1h = hmc.I_0_2(cosmo, k_use, aa, prof,
-                              prof_2pt, prof2)
+            pk_1h = hmc.I_0_2(cosmo, k_use, aa, prof, prof_2pt, prof2)
+            if supress_1h is not None:
+                ks = supress_1h(aa)
+                pk_1h *= (k_use / ks)**4 / (1 + (k_use / ks)**4)
         else:
             pk_1h = 0.
 
-        # Total power spectrum
-        out[ia, :] = (pk_1h + pk_2h) * norm
+        # Transition region
+        if smooth_transition is None:
+            out[ia, :] = (pk_1h + pk_2h) * norm
+        else:
+            alpha = smooth_transition(aa)
+            if alpha < 0.4 and not flag_alpha:
+                flag_alpha = True
+            out[ia, :] = (pk_1h**alpha + pk_2h**alpha)**(1/alpha) * norm
+
+    if flag_alpha:
```

Comment:
I am not sure we need this warning. What's the rationale?

Reply:
Here in the figure, I'm plotting P(k, a=1) for the autocorrelation of the NFW profile, for the smoothing parameter ranging in 0.2 < α < 2.
For α > 1 the sharpening does not behave in a weird manner, but smoothing (for α < 1) only makes sense up until α ~ 0.4. After that it doesn't really smooth the transition, rather, it affects the entire P(k). I could have been stricter and imposed a warning for α < 0.5, but I figured there might be other profiles where α being so low might make sense.
![alphas](https://user-images.githubusercontent.com/22447385/120923648-62ef0800-c6d8-11eb-9934-bb1a3ede2199.png)

Reply:
The keyword is `smooth_transition` anyway. The point is that users shouldn't expect that only the transition changes for any α < 0.4.

Reply:
And here it the equivalent plot for P(k, a=0.3).
![alphas_2](https://user-images.githubusercontent.com/22447385/120923991-245a4d00-c6da-11eb-91f9-5205feccb2c6.png)

Reply:
And here's the same thing for the HOD profile's autocorrelation (using the default HOD params) for scale factor 0.3.
![HOD](https://user-images.githubusercontent.com/22447385/120924111-b06c7480-c6da-11eb-9cc4-b2903f54bfa8.png)

Reply:
And lastly, for the GNFW profile (using the default params) and scale factor 0.3.
![GNFW](https://user-images.githubusercontent.com/22447385/120924131-cb3ee900-c6da-11eb-9f2e-8b786211e550.png)

Reply:
Ok, thanks.
So alpha<0.4 may not seem sensible to you (or to me), but it's still a valid operation someone might want to perform. For instance, one may want to marginalize over alpha in an MCMC, which will often cross this boundary.
I'd much rather not burden the code with extra lines and checks in this case, so I'd remove this check altogether.

Reply:
Yep, that makes sense. I'll remove it. Thanks!

Reply:
Done.